File: ObjcExample/ObjcExample/Demo/GKPageScrollView/抖音/GKDYListViewController.m
//
//  GKDYListViewController.m
//  GKPageScrollView
//
//  Created by QuintGao on 2018/10/28.
//  Copyright © 2018 QuintGao. All rights reserved.
//

#import "GKDYListViewController.h"

@interface GKDYListViewController ()

@property (nonatomic, copy) void(^listScrollViewScrollBlock)(UIScrollView *scrollView);

@end

@implementation GKDYListViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.gk_navigationBar.hidden = YES;
    
    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:@"listCell"];
}

#pragma mark - UITableViewDataSource & UITableViewDelegate
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return 30;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"listCell" forIndexPath:indexPath];
    cell.selectionStyle = UITableViewCellSelectionStyleNone;
    cell.textLabel.text = [NSString stringWithFormat:@"第%zd行", indexPath.row + 1];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    UIViewController *vc = [[UIViewController alloc] init];
    vc.view.backgroundColor = UIColor.whiteColor;
    vc.gk_navTitle = @"testVC";
    [self.navigationController pushViewController:vc animated:YES];
}

#pragma mark - UIScrollViewDelgate
- (void)scrollViewDidScroll:(UIScrollView *)scrollView {
    !self.listScrollViewScrollBlock ? : self.listScrollViewScrollBlock(scrollView);
}

#pragma mark - GKPageListViewDelegate
- (UIScrollView *)listScrollView {
    return self.tableView;
}

- (void)listViewDidScrollCallback:(void (^)(UIScrollView * _Nonnull))callback {
    self.listScrollViewScrollBlock = callback;
}

- (UIView *)listView {
    return self.view;
}

- (void)listWillAppear {
    NSLog(@"%zd--%s", self.index, __func__);
}

- (void)listDidAppear {
    NSLog(@"%zd--%s", self.index, __func__);
}

- (void)listWillDisappear {
    NSLog(@"%zd--%s", self.index, __func__);
}

- (void)listDidDisappear {
    NSLog(@"%zd--%s", self.index, __func__);
}

@end

File: SwiftExample/SwiftExample/Demo/GKPageScrollView/抖音/GKDYListViewController.swift
//
//  GKDYListViewController.swift
//  GKPageScrollViewSwift
//
//  Created by gaokun on 2019/2/26.
//  Copyright © 2019 gaokun. All rights reserved.
//

import UIKit
import GKPageScrollView

class GKDYListViewController: GKBaseTableViewController {

    var scrollCallBack: ((UIScrollView) -> ())?
    var index: Int = 0
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.gk_navigationBar.isHidden = true
        
        self.tableView.register(UITableViewCell.classForCoder(), forCellReuseIdentifier: "listCell")
    }
}

extension GKDYListViewController {
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 30
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "listCell", for: indexPath)
        cell.selectionStyle = .none
        cell.textLabel?.text = "第" + "\(indexPath.row + 1)" + "行"
        return cell
    }
    
    func scrollViewDidScroll(_ scrollView: UIScrollView) {
        scrollCallBack!(scrollView)
    }
}

extension GKDYListViewController: GKPageListViewDelegate {
    func listScrollView() -> UIScrollView {
        self.tableView
    }
    
    func listViewDidScroll(callBack: @escaping (UIScrollView) -> ()) {
        self.scrollCallBack = callBack
    }
    
    func listView() -> UIView {
        self.view
    }
    
    func listWillAppear() {
        print("\(index)===listWillAppear")
    }
    
    func listDidAppear() {
        print("\(index)===listDidAppear")
    }
    
    func listWillDisappear() {
        print("\(index)===listWillDisappear")
    }
    
    func listDidDisappear() {
        print("\(index)===listDidDisappear")
    }
}
